watidl/watidl.md

```diff
@@ -0,0 +1,119 @@
+# WatIDL :duck:
+
+
+WatIDL (`/ˈwädl/`) is an IDL for webassembly. Its goal is to be able to fully describe the import and export capabilities of Webassembly modules while being intuitive to read and reason about. It inherits the Webassembly text format's syntax and attempts to closely follow and fit into the Webassembly ecosystem. WatIDL should used as a simple syntax for describing the interfaces along with WebIDL bindings to describe how the different data structures are formed and received by Wasm modules.
+
+# Differences from WAT
+While WatIDL follow close to WAT there are some differences
+
+## Types
+WatIDL uses the following primitive types
+
+```
+intType := "i8" | "i6" | "i32" | "i64" | "u8" | "u16" | "u32" | "u64";
```

Comment:
In underlying wasm, `i32` and `i64` refer to types which don't have a signedness interpretation. Using them to indicate signed types here could lead to ambiguity in some contexts. I suggest using `s32` and similar names for explicitly signed types.

Reply:
updated! thanks